doc/changelog.md

```diff
@@ -1,4 +1,37 @@
-# Release 0.4.0
+# Release 0.4.1
+
+<h3>Bug fixes</h3>
+
+* Resolve a failure to generate gradient code for specific input circuits.
+  [(#439)](https://github.com/PennyLaneAI/catalyst/issues/439)
+
+  In this case, [`jnp.mod`](https://github.com/PennyLaneAI/catalyst/issues/437)
+  was used to compute wire values in a for loop, which prevented the gradient
+  architecture from fully separating quantum and classical code. The following
+  program is now supported:
+  ```py
+  @qjit
+  @grad
+  @qml.qnode(dev)
+  def f(x):
+      def cnot_loop(j):
+          qml.CNOT(wires=[j, jnp.mod((j + 1), 4)])
+  
+      for_loop(0, 4, 1)(cnot_loop)()
+  
+      return qml.expval(qml.PauliZ(0))
+  ```
+
+* Resolve a memory leak in the runtime stemming from  missing calls to device destructors
+  at the end of programs.
+  [(#446)](https://github.com/PennyLaneAI/catalyst/pull/446)
```

Comment:
@maliasadi @dime10 I only cherry picked in commit 8f66c16. Do you know why this PR is also being pulled in? Is it just in the changelog, or below in the logic as well?